lua/diffview/vcs/adapters/hg/init.lua

```diff
@@ -1,18 +1,1051 @@
+      adapter = self,
+      path = v.name,
+      oldpath = v.oldname,
+      status = v.status,
+      stats = v.stats,
+      kind = kind,
+      revs = {
+        a = left,
+        b = right,
+      }
+    }))
+  end
+
+  callback(nil, files, conflicts)
+end, 7)
+
+HgAdapter.untracked_files = async.wrap(function(self, left, right, opt, callback)
+  Job:new({
+    command = self:bin(),
+    args = utils.vec_join(
+      self:args(),
+      "status",
+      "--print0",
+      "--unknown",
+      "--no-status",
+      "--template={path}\\n"
+    ),
+    cwd = self.ctx.toplevel,
+    ---@type Job
+    on_exit = function(j)
+      utils.handle_job(j, {
+        debug_opt = {
+          context = "HgAdapter>untracked_files()",
+          func = "s_debug",
+          debug_level = 1,
+          no_stdout = true,
+        },
+      })
+
+      if j.code ~= 0 then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      local files = {}
+      for _, s in ipairs(j:result()) do
+        table.insert(
+          files,
+          FileEntry.with_layout(opt.default_layout, {
+            adapter = self,
+            path = s,
+            status = "?",
+            kind = "working",
+            revs = {
+              a = left,
+              b = right,
+            }
+          })
+        )
+      end
+      callback(nil, files)
+    end,
+  }):start()
+end, 5)
+
+---@param self HgAdapter
+---@param path string
+---@param rev? Rev
+---@param callback fun(stderr: string[]?, stdout: string[]?)
+HgAdapter.show = async.wrap(function(self, path, rev, callback)
+  -- File did not exist, need to return an empty buffer
+  if not(rev) or (rev:object_name() == self.Rev.NULL_TREE_SHA) then
+    callback(nil, {})
+    return
+  end
+
+  local job = Job:new({
+    command = self:bin(),
+    args = self:get_show_args(path, rev),
+    cwd = self.ctx.toplevel,
+    ---@type Job
+    on_exit = async.void(function(j)
+      local context = "HgAdapter.show()"
+      utils.handle_job(j, {
+        fail_on_empty = true,
+        context = context,
+        debug_opt = { no_stdout = true, context = context },
+      })
+
+      if j.code ~= 0 then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      local out_status
+
+      if #j:result() == 0 then
+        async.util.scheduler()
+        out_status = vcs_utils.ensure_output(2, { j }, context)
+      end
+
+      if out_status == JobStatus.ERROR then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      callback(nil, j:result())
+    end),
+  })
+  -- Problem: Running multiple 'show' jobs simultaneously may cause them to fail
+  -- silently.
+  -- Solution: queue them and run them one after another.
+  vcs_utils.queue_sync_job(job)
+end, 4)
+
+HgAdapter.flags = {
+  ---@type FlagOption[]
+  switches = {
+    {'--follow', '-f', 'Follow renames'},
+    {'--no-merges', '-M', 'List no merge changesets'},
```

Comment:
The order should be `keymap`, `flag_name`, `description`. So:

```suggestion
    { '-f', '--follow', 'Follow renames' },
    { '-M', '--no-merges', 'List no merge changesets' },
```